### PR TITLE
for issue 232

### LINKED ISSUE #232: API calls missing the response status
I am integrating with API calls to APIG and it seems like the response status is missing.
The response object comes back as an empty object. 

I am invoking the call as in the official docs

```
API.post(apiName, path, myInit).then(response => {
    if(response) {
     this.setState({sucess:true})
    } else {
      this.setState({sucess:false})
    }
});
```
If the request success with a 2XX I can parse the response array just fine.
However, when the request results in an error, there is no response status to parse, e.g. 4xx, 5xx.
I can see in Chrome dev tools that the status codes are indeed present.

Any ideas?

using 
- aws-amplify: 0.2.3
- aws-amplify-react: 0.1.32

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
Fixes #232 

*Description of changes:*
Return full response of api if you set ```response``` in the ```init``` object tot `true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/media/api_guide.md [hunk 1]
```diff
@@ -85,6 +85,8 @@ The below code assumes use of the Automated Setup.
 
 Each method of Amplify's API module returns a Promise which is seen in the below examples with different HTTP verbs. Configure the `apiName`, `path` and `headers` according to your settings.
 
+Note: To get the full response from API call, set ```returnAll``` to ```true``` in the ```init``` object. That's a temporary solution, we will deprecate the old version in the future.
```
Comment (mlabieniec) [inline docs/media/api_guide.md]: Update:

Note: To retrieve the full response from the API set `response` to `true` in the `init` object.

Comment (powerful23) [reply docs/media/api_guide.md]: What about ```responseAll```? ```response``` is a little confusing since we do have a response for these methods.

Comment (mlabieniec) [reply docs/media/api_guide.md]: Well my thought is that we do have a response but we are not returning it, we are returning the data object. So setting response true is saying actually return me the response not the data object...

Comment (powerful23) [reply docs/media/api_guide.md]: Gotcha.

File: docs/media/api_guide.md [hunk 2]
```diff
@@ -85,13 +85,16 @@ The below code assumes use of the Automated Setup.
 
 Each method of Amplify's API module returns a Promise which is seen in the below examples with different HTTP verbs. Configure the `apiName`, `path` and `headers` according to your settings.
 
+Note: To get the full response from API call, set ```response``` to ```true``` in the ```init``` object. That's a temporary solution, we will deprecate the old version in the future.
```
Comment (mlabieniec) [inline docs/media/api_guide.md]: I don't think we need the last sentence (temporary part), we can just have the first there for now.

Comment (powerful23) [reply docs/media/api_guide.md]: updated.

**Pull request merged**